R/add_cols_result_df_pd_changes.R

```diff
@@ -0,0 +1,61 @@
+#' Function that renames Merton model baseline result columns and adds additional
+#' columns for the second application of the model in the late & sudden scenario.
+#'
+#' @param data A dataframe that contains the result data from the baseline
+#'   Merton model
+#' @param horizon A cahracter vector of length 1 indicating whether to return
+#'   a result obejct for overall or yearly results. Must be "annual" or "overall"
+#'
+#' @family helper functions
+#'
+#' @export
+
+add_cols_result_df_pd_changes <- function(data,
+                                          horizon = NULL) {
+
+  force(data)
+  horizon %||% stop("Must provide input for 'horizon'", call. = FALSE)
+
+  horizon_allowed <- horizon %in% c(
+    "annual",
+    "overall"
+  )
+  stopifnot(horizon_allowed)
+
+  if (horizon == "overall") {
+    data_has_expected_columns <- all(
+      c(
+        "investor_name", "portfolio_name", "scenario_name", "scenario_geography",
+        "id", "company_name", "ald_sector", "technology", "equity_0_baseline",
+        "equity_0_late_sudden", "debt", "volatility", "risk_free_rate", "term",
+        "Maturity", "Vt", "St", "Dt", "Survival"
+      ) %in% colnames(data)
+    )
+  } else {
+    data_has_expected_columns <- all(
+      c(
+        "investor_name", "portfolio_name", "scenario_name", "scenario_geography",
+        "id", "company_name", "ald_sector", "technology", "year",
+        "equity_t_baseline", "equity_t_late_sudden", "debt", "volatility",
+        "risk_free_rate", "term", "Maturity", "Vt", "St", "Dt", "Survival"
+      ) %in% colnames(data)
+    )
```

Comment:
You may DRY this duplication and get a clearer overview of what's the difference between these two.

Maybe

```r
common <- c("a", "b")
expected <- c(common, "d")
if (<something happens>) {
  expected <- c(common, "f")
}
expected

Reply:
This is a nice idea!